portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,31 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import com.google.gson.Gson;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> comments = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Karen!</h1>");
+    String json = new Gson().toJson(comments);
+    response.setContentType("application/json");
+    response.getWriter().write(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String newComment = request.getParameter("user-comment");
+    if (newComment.isEmpty()) {
+      System.err.println("Unfortunately, you didn't write anything.");
+    } else {
+      comments.add(newComment);
```

Comment:
You can extract this from the else (that is, remove the else completely). This is what's called a happy path (https://en.wikipedia.org/wiki/Happy_path):

if (newComment.isEmpty()) { // throw some error }
// Rest of execution

Reply:
got it, thanks!